packages/valory/skills/decision_maker_abci/behaviours/sell_outcome_token.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2024 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+
+"""This module contains the behaviour for selling a token."""
+from typing import Any, Generator, Optional, cast
+
+from hexbytes import HexBytes
+
+from packages.valory.contracts.market_maker.contract import (
+    FixedProductMarketMakerContract,
+)
+from packages.valory.protocols.contract_api import ContractApiMessage
+from packages.valory.skills.decision_maker_abci.behaviours.base import (
+    DecisionMakerBaseBehaviour,
+    WaitableConditionType,
+)
+from packages.valory.skills.decision_maker_abci.models import MultisendBatch
+from packages.valory.skills.decision_maker_abci.payloads import MultisigTxPayload
+from packages.valory.skills.decision_maker_abci.states.sell_outcome_token import (
+    SellOutcomeTokenRound,
+)
+
+
+class SellTokenBehaviour(DecisionMakerBaseBehaviour):
+    """A behaviour in which the agents sell a token."""
+
+    matching_round = SellOutcomeTokenRound
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize the sell token behaviour."""
+        super().__init__(**kwargs)
+        self.sell_amount: float = 0.0
+        self.return_amount: int = 0
+
+    @property
+    def market_maker_contract_address(self) -> str:
+        """Get the contract address of the market maker on which the service is going to place the bet."""
+        return self.sampled_bet.id
+
+    @property
+    def outcome_index(self) -> int:
+        """Get the index of the outcome for which the service is going to sell token."""
+        return cast(int, self.synchronized_data.vote)
+
+    def _build_approval_tx(self) -> WaitableConditionType:
+        """Build an ERC20 approve transaction."""
+        approval_tx = yield from self.build_approval_tx(
+            self.return_amount,
+            self.market_maker_contract_address,
+            self.market_maker_contract_address,
+        )
+        return approval_tx
```

Comment:
The approval tx is stored in the `multisend_batches` attribute as a batch. The method returns a status of success.

```suggestion
        status = yield from self.build_approval_tx(
            self.return_amount,
            self.market_maker_contract_address,
            self.market_maker_contract_address,
        )
        return status
```